pkg/restapi/rp/operation/operations.go

```diff
@@ -168,22 +195,39 @@ func (o *Operation) hydraLoginHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (o *Operation) setLoginRequestForState(state string, request *models.LoginRequest) {
-	o.lock.Lock()
-	defer o.lock.Unlock()
+	o.oidcStateLock.Lock()
+	defer o.oidcStateLock.Unlock()
 
 	o.oidcStates[state] = request
 }
 
 func (o *Operation) getAndUnsetLoginRequest(state string) *models.LoginRequest {
-	o.lock.Lock()
-	defer o.lock.Unlock()
+	o.oidcStateLock.Lock()
+	defer o.oidcStateLock.Unlock()
 
 	r := o.oidcStates[state]
 	delete(o.oidcStates, state)
 
 	return r
 }
 
+func (o *Operation) setConsentRequest(handle string, r *consentRequest) {
+	o.presDefsLock.Lock()
+	defer o.presDefsLock.Unlock()
+
+	o.consentRequests[handle] = r
+}
+
+func (o *Operation) getAndUnsetConsentRequest(handle string) *consentRequest {
+	o.presDefsLock.Lock()
+	defer o.presDefsLock.Unlock()
+
+	r := o.consentRequests[handle]
```

Comment:
Assuming this would be stored in some persistent store rather than in-memory.

Reply:
yes